### PR TITLE
[ADF-3666] User Profile - Provide a way to show SSO Logged-In User

### PR DESCRIPTION
**Pull request by sivakumar414ram**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [x] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://issues.alfresco.com/jira/browse/ADF-3666

**What is the new behaviour?**

https://issues.alfresco.com/jira/browse/ADF-3666

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/userinfo/components/user-info.component.html [hunk 1]
```diff
@@ -1,48 +1,68 @@
-<div id="userinfo_container"
-    [class.adf-userinfo-name-right]="showOnRight()"
+<div id="userinfo_container" [class.adf-userinfo-name-right]="showOnRight()"
     class="adf-userinfo-container" *ngIf="isLoggedIn()">
-    <span *ngIf="ecmUser && showName" id="adf-userinfo-ecm-name-display"
-          class="adf-userinfo-name">{{ecmUser.fullNameDisplay}}</span>
-    <span *ngIf="bpmUser && !ecmUser && showName" id="adf-userinfo-bpm-name-display"
-          class="adf-userinfo-name">{{bpmUser.fullNameDisplay}}</span>
+
+    <div *ngIf="showName">
```
Comment (mauriziovitale) [inline lib/core/userinfo/components/user-info.component.html]: can we use ng content ?

File: lib/core/services/jwt-helper.service.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        let decoded = this.urlBase64Decode(parts[1]);
+        if (!decoded) {
+            throw new Error('Cannot decode the token');
+        }
+
+        return JSON.parse(decoded);
+    }
+
+    private urlBase64Decode(token) {
```
Comment (eromano) [inline lib/core/services/jwt-helper.service.ts]: missing return type

File: lib/core/userinfo/components/user-info.component.ts [hunk 3]
```diff
@@ -72,58 +75,47 @@ export class UserInfoComponent implements OnInit {
...
+        } else if (this.authService.isBpmLoggedIn()) {
+            this.loadBpmUserInfo();
+        }
     }
 
     isLoggedIn() {
         return this.authService.isLoggedIn();
     }
 
     loadEcmUserInfo(): void {
```
Comment (eromano) [inline lib/core/userinfo/components/user-info.component.ts]: void return type not needed

File: lib/core/userinfo/components/user-info.component.ts [hunk 4]
```diff
@@ -72,58 +75,47 @@ export class UserInfoComponent implements OnInit {
...
+            this.loadEcmUserInfo();
+            this.loadBpmUserInfo();
+        } else if (this.authService.isEcmLoggedIn()) {
+            this.loadEcmUserInfo();
+        } else if (this.authService.isBpmLoggedIn()) {
+            this.loadBpmUserInfo();
+        }
     }
 
     isLoggedIn() {
```
Comment (eromano) [inline lib/core/userinfo/components/user-info.component.ts]: missing return type

File: lib/core/userinfo/services/ecm-user.service.ts [hunk 5]
```diff
@@ -42,7 +42,9 @@ export class EcmUserService {
     getUserInfo(userName: string): Observable<EcmUserModel> {
         return from(this.apiService.getInstance().core.peopleApi.getPerson(userName))
             .pipe(
-                map(data => <EcmUserModel> data['entry']),
+                map((data) => {
+                    return new EcmUserModel(data['entry']);
```
Comment (eromano) [inline lib/core/userinfo/services/ecm-user.service.ts]: why do data['entry'] and not just data.entry? and data which kind of type is it?

File: lib/core/services/jwt-helper.service.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
+/*!
```
Comment (eromano) [inline lib/core/services/jwt-helper.service.ts]: this class doesn't have any test

File: lib/core/userinfo/services/identity-user.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+            CoreModule.forRoot()
+        ]
+    });
+
+    beforeEach(() => {
+        service = TestBed.get(IdentityUserService);
+    });
+
+    it('should able to fetch identity user info from Jwt token', (done) => {
+        spyOn(service, 'getCurrentUserInfo').and.returnValue(of({firstName: 'fake-first-name', lastName: 'fake-last-name', email: 'fake@gmail.com'}));
```
Comment (eromano) [inline lib/core/userinfo/services/identity-user.service.spec.ts]: You are not passing any JWT in this way you are not testing it

File: lib/core/services/jwt-helper.service.ts (hunk 6 above)
Comment (sivakumar414ram) [reply lib/core/services/jwt-helper.service.ts]: Added unit test cases

File: lib/core/userinfo/services/identity-user.service.spec.ts (hunk 7 above)
Comment (sivakumar414ram) [reply lib/core/userinfo/services/identity-user.service.spec.ts]: JWT is used to test

File: lib/core/userinfo/services/ecm-user.service.ts (hunk 5 above)
Comment (sivakumar414ram) [reply lib/core/userinfo/services/ecm-user.service.ts]: the return type is an Object  '{}'
if we use data.entry, it leads to this error: Property 'entry' does not exist on type {}

File: lib/core/userinfo/components/user-info.component.html (hunk 1 above)
Comment (sivakumar414ram) [reply lib/core/userinfo/components/user-info.component.html]: Done Used ng-container

File: lib/core/userinfo/services/ecm-user.service.ts (hunk 5 above)
Comment (eromano) [reply lib/core/userinfo/services/ecm-user.service.ts]: Dot notation is much easier to read than bracket notation I would  prefer to use it as in the rest of the code.   Property 'entry' does not exist on type {} means that data is empty maybe we need to check for it. I guess the type of data should be PersonEntry and please change the name of the variable data from 

```
(data)  to (personEntry:PersonEntry)
```

Comment (mauriziovitale) [reply lib/core/userinfo/services/ecm-user.service.ts]: so this 
`new EcmUserModel(data['entry']);`
will become 
`new EcmUserModel(person.entry]);`

**Pull request merged**